t/25-cache.t

```diff
@@ -0,0 +1,247 @@
+#! /usr/bin/perl
+
+# Copyright (c) 2016 SUSE LINUX GmbH, Nuernberg, Germany.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+BEGIN {
+    unshift @INC, 'lib';
+    $ENV{OPENQA_TEST_IPC} = 1;
+}
+
+use strict;
+use warnings;
+
+use Test::More;
+use Test::Warnings;
+use Test::MockModule;
+use OpenQA::Utils;
+use OpenQA::Worker::Cache;
+use DBI;
+use File::Path qw(remove_tree make_path);
+use File::Spec::Functions qw(catdir catfile);
+use Digest::MD5 qw(md5);
+use Cwd qw(abs_path getcwd);
+
+use Mojo::IOLoop;
+use Mojolicious;
+use Mojo::Server::Daemon;
+use Mojolicious::Lite;
+
+
+
+# create Test DBus bus and service for fake WebSockets call
+# my $ws = OpenQA::WebSockets->new;
+
+my $sql;
+my $sth;
+my $result;
+my $dbh;
+my $filename;
+
+my $cachedir = catdir(getcwd(), 't/cache.d/cache');
+my $db_file = "$cachedir/cache.sqlite";
+$ENV{LOGDIR} = catdir(getcwd(), 't/cache.d', 'logs');
+my $logfile = catdir($ENV{LOGDIR}, 'cache.log');
+my $host = 'http://localhost:8080';
+
+remove_tree($cachedir);
+make_path($ENV{LOGDIR});
+
+#redirect stdout
+open(my $FD, '>>', $logfile);
+select $FD;
+$| = 1;
+truncate_log();
+# Mock of log_* methods used by Cache:
+
+my $openqa_utils = new Test::MockModule('OpenQA::Utils');
+#log_error log_info log_debug
+
+$openqa_utils->mock(log_info  => \&mock_log_info);
+$openqa_utils->mock(log_error => \&mock_log_error);
+$openqa_utils->mock(log_debug => \&mock_log_debug);
+$openqa_utils->mock(delete    => \&mock_delete);
```

Comment:
I don't understand why you need to mock logging functions. Aren't your fallbacks what lib/OpenQA/Utils.pm already provides? Or do you have problems with the redirection to stderr instead of stdout?